app/Http/Controllers/HoldController.php

```diff
@@ -169,11 +170,14 @@ public function assignHold(Request $request): JsonResponse
             return response()->json([], 422);
         }
 
-        $networkAircraft = NetworkAircraft::firstOrCreate(['callsign' => $request->json('callsign')]);
+        NetworkDataService::firstOrCreateNetworkAircraft(
+            $request->json('callsign'),
+            ['callsign' => $request->json('callsign')]
+        );
         $assignedHold = AssignedHold::updateOrCreate(
             ['callsign' => $request->json('callsign')],
             [
-                'callsign' => $networkAircraft->callsign,
+                'callsign' => $request->json('callsign'),
```

Comment:
Would it be better to assign the model creation to a variable and then do `->callsign` like before - I am thinking this might minimise repetition / required changes if we end up changing in the input name? Either that or assign `$callsign = $request->json('callsign')` and reference that everywhere?

Reply:
Done!